src/java/org/apache/cassandra/db/compression/CompressionDictionaryManager.java

```diff
@@ -267,6 +312,12 @@ private void handleNewDictionary(CompressionDictionary dictionary)
         onNewDictionaryTrained(dictionary.dictId());
     }
 
+    // TODO Trying to figure out how to do this most effectively
+    public void importDictionary(CompressionDictionary dictionary)
```

Comment:
I am not completely sure this is right and is enough but seems like it. The dict is stored in distributed table eventually.

I am not completely sure what should happen if we e.g. insert dictionary which is there already. (exact same dict id, keyspace and table name). I think that we should check before by retrieving it first and if not there insert. If it is there already we should most probably fail.